src/components/TagContainer.ts

```diff
@@ -0,0 +1,237 @@
+    }
+
+    private getValue(attribute: string, mxObject?: mendix.lib.MxObject): string {
+        return mxObject ? (mxObject.get(attribute) as string) : "";
+    }
+
+    private isReadOnly() {
+        const { editable, mxObject, readOnly } = this.props;
+        if (editable === "default" && mxObject) {
+            return readOnly || mxObject.isReadonlyAttr(this.tagAttribute);
+        }
+
+        return true;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.tagAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions() {
+        this.setState({
+            newTag: this.getValue(this.tagAttribute, this.props.mxObject) as string
+        });
+    }
+
+    private lazyLoadTags() {
+        if (this.props.mxObject) {
+            this.fetchCurrentTags(this.props.mxObject);
+        }
+    }
+
+    private fetchCurrentTags(mxObject: mendix.lib.MxObject) {
+        const { tagConstraint } = this.props;
+        const constraint = tagConstraint
+            ? tagConstraint.replace(/\[\%CurrentObject\%\]/gi, mxObject.getGuid())
+            : "";
+        const XPath = "//" + this.tagEntity + constraint;
+        mx.data.get({
+            callback: object => {
+                const suggestionList = object.map(tagObject => ({
+                    value: tagObject.get(this.tagAttribute) as string
+                }));
+                this.setState({ suggestions: suggestionList.map(suggestion => suggestion.value) });
+                this.processTags(object);
+            },
+            error: error =>
+                window.mx.ui.error(`An error occurred while retrieving tags via XPath (${this.tagEntity}): ${error}`),
+            xpath: XPath
+        });
+    }
+
+    private processTags(tagObjects: any[]) {
+        const currentTagObjects: mendix.lib.MxObject[] = [];
+        const referenceTags = this.props.mxObject.get(this.referenceAttribute) as string[];
+        tagObjects.map(object => {
+            referenceTags.map(reference => {
+                if (reference === object.getGuid()) {
+                    currentTagObjects.push(object);
+                }
+            });
+        });
+        const tagData = currentTagObjects.map(tagObject => ({ value: tagObject.get(this.tagAttribute) as string }));
+        this.setState({
+            tagList: tagData.map(tag => tag.value)
+        });
+    }
+
+    private createTag(tag: string) {
+        const { afterCreateMicroflow, mxObject } = this.props;
+        mx.data.create({
```

Comment:
Dont blindly create a new tag, what if it already exists? First check the suggestion list, if not available check the get by `xpath://${this.tagEntity}[ ${this.tagAttribute} = '${name}' ]`  if not found at all.. create a new one

Reply:
Fixed